### PR TITLE
feat: add loss for CRNN

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR implements compute_target, compute_loss, and corresponding unitests for CRNN model.
Any feedback is welcome!

### CODE REVIEW

File: doctr/models/recognition/crnn.py [hunk 1]
```diff
@@ -70,6 +74,54 @@ def __init__(
...
+        encoded = encode_sequences(
+            sequences=gts,
+            vocab=self.vocab,
+            target_size=self.feat_extractor.output_shape[2],
+            eos=len(self.vocab)
+        )
+        tf_encoded = tf.cast(encoded, tf.int64)
+        seq_len = [len(word) for word in gts]
+        tf_seq_len = tf.cast(seq_len, tf.int64)
+        return tf_encoded, tf_seq_len
```
Comment (fg-mindee) [inline doctr/models/recognition/crnn.py:99]: Let's move this to recognition/core.py in RecognitionModel

File: test/test_models_recognition.py [hunk 2]
```diff
@@ -164,3 +164,23 @@ def test_compute_loss_sar(test_compute_target_sar):
...
+
+@pytest.fixture(scope="function")
+def test_compute_target_crnn():
+    list_gts = ['elephants', '1234', 'Rémouleur']
+    model = recognition.crnn_vgg16_bn()
+    encoded_gts, seq_len = model.compute_target(list_gts)
+    assert isinstance(encoded_gts, tf.Tensor)
+    assert isinstance(seq_len, tf.Tensor)
+    assert list(seq_len.numpy()) == [9, 4, 9]
+    return encoded_gts, seq_len
```
Comment (fg-mindee) [inline test/test_models_recognition.py:177]: We will be able to refactor this test with the SAR if we implement my above comment

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR! Only a small refactor to improve the design

File: test/test_models_recognition.py [hunk 3]
```diff
@@ -146,41 +146,27 @@ def test_recognition_zoo_error():
...
-
-
-@pytest.fixture(scope="function")
-def test_compute_target_crnn():
-    list_gts = ['elephants', '1234', 'Rémouleur']
-    model = recognition.crnn_vgg16_bn()
-    encoded_gts, seq_len = model.compute_target(list_gts)
     assert isinstance(encoded_gts, tf.Tensor)
     assert isinstance(seq_len, tf.Tensor)
     assert list(seq_len.numpy()) == [9, 4, 9]
```
Comment (fg-mindee) [inline test/test_models_recognition.py:159]: Let's move this tests before computing the loss (no need if this fails)

File: test/test_models_recognition.py [hunk 4]
```diff
@@ -146,41 +146,27 @@ def test_recognition_zoo_error():
...
     model_input = tf.random.uniform(shape=[3, 32, 128, 3], minval=0, maxval=1)
     model = recognition.crnn_vgg16_bn()
     model_output = model(model_input)
-    encoded_gts, seq_len = test_compute_target_crnn
+    encoded_gts, seq_len = model.compute_target(list_gts)
     loss = model.compute_loss(encoded_gts, model_output, seq_len)
     assert isinstance(loss, tf.Tensor)
+    assert isinstance(encoded_gts, tf.Tensor)
+    assert isinstance(seq_len, tf.Tensor)
+    assert list(seq_len.numpy()) == [9, 4, 9]
```
Comment (fg-mindee) [inline test/test_models_recognition.py:172]: Same here

**Review by fg-mindee — COMMENTED** [review]
Another improvement suggestion in comment!

**Review by fg-mindee — APPROVED** [review]
Thanks for the edits!
